src/parser/view_transform.c

```diff
@@ -3199,6 +3198,14 @@ mq_copypush_sargable_terms_helper (PARSER_CONTEXT * parser, PT_NODE * statement,
 
   for (term = statement->info.query.q.select.where; term; term = term->next)
     {
+      /* check for on_cond term */
+      assert (term->node_type == PT_EXPR || term->node_type == PT_VALUE);
+      if ((term->node_type == PT_EXPR && term->info.expr.location > 0)
+	  || (term->node_type == PT_VALUE && term->info.expr.location > 0))
+	{
```

Comment:
If the node_type is PT_VALUE the info.expr is not valid but th info.value.... is available. Please check if it's correct.

Reply:
thanks. I fixed it.